vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -308,8 +311,15 @@ public void setLocale(Locale locale) {
     }
 
     private void setLocaleWithJS() {
-        runBeforeClientResponse(ui -> getElement()
-                .callJsFunction("$connector.setLocale", languageTag));
+        runBeforeClientResponse(ui -> {
+            boolean hasDateFormats = i18n != null && i18n.dateFormats != null
+                    && i18n.dateFormats.size() > 0;
+
+            if (!hasDateFormats) {
```

Comment:
We might want to add a comment explaining why we prevent setting the locale in this case. The only hint at the mechanism that the custom format has higher prio than the locale, is a single line in the DatePickerI18n javadoc.

Reply:
added a simple comment, I'm not sure if it suffices